### DISCUSSION TITLE
Is it possible, for development purposes, to replace vcpkg_from_git() in the portfile with some pointer to a local directory?

### DISCUSSION THREAD

**Discussion by McKillroy** [microsoft/vcpkg#20178] [Q&A] [answered]
For development of a port I'd like to have the library sources in a local directory until everything works, 
and then later push them to their git repo. 
In the meantime I'd like to alter the portfile in the custom registry to use the local directory, 
later I want to replace that with vcpkg_from_git().

In the moment I am trying to use an URL like `URL file:///C:/...` inside  vcpkg_from_git(), but it still wants a ref,
which forces me each time I change something to commit/amend and get the ref with git show-ref, which is kinda cumbersome.

Is there a better way to use a local directory for the ports sources?

**Comment by jsanchez19** [comment, marked as answer]
for testing, what I do is simply comment out the `vcpkg_from_git()` call and set the `SOURCE_PATH` variable directly:

```cmake
set(SOURCE_PATH "C:/Users/user/path/to/local/project")
#vcpkg_from_git(
#    OUT_SOURCE_PATH SOURCE_PATH
#    URL https://github.com/asdf/asdf.git
#    REF ...
#)

```

**Comment by McKillroy** [comment]
That worked! 
Thanks a lot @jsanchez19 !
